sklearn/ensemble/tests/test_forest.py

```diff
@@ -1330,3 +1330,96 @@ def test_forest_degenerate_feature_importances():
     gbr = RandomForestRegressor(n_estimators=10).fit(X, y)
     assert_array_equal(gbr.feature_importances_,
                        np.zeros(10, dtype=np.float64))
+
+
+@pytest.mark.parametrize('name', FOREST_CLASSIFIERS_REGRESSORS)
+@pytest.mark.parametrize(
+    'max_samples, exc_type, exc_msg',
+    [(int(1e9), ValueError,
+      "`max_samples` must be in range 1 to 6 but got value 1000000000"),
+     (1.0, ValueError,
+      "`max_samples` must be in range (0, 1) but got value 1.0"),
+     (2.0, ValueError,
+      "`max_samples` must be in range (0, 1) but got value 2.0"),
+     (0.0, ValueError,
+      "`max_samples` must be in range (0, 1) but got value 0.0"),
+     (np.nan, ValueError,
+      "`max_samples` must be in range (0, 1) but got value nan"),
+     (np.inf, ValueError,
+      "`max_samples` must be in range (0, 1) but got value inf"),
+     ('str max_samples?!', TypeError,
+      "`max_samples` should be int or float, but got type '<class \'str\'>'"),
+     (np.ones(2), TypeError,
+      "`max_samples` should be int or float, but got type "
+      "'<class \'numpy.ndarray\'>'")]
+)
+def test_max_samples_exceptions(name, max_samples, exc_type, exc_msg):
+    # Check invalid `max_samples` values
+    est = FOREST_CLASSIFIERS_REGRESSORS[name](max_samples=max_samples)
+    with pytest.raises(exc_type) as exc_info:
+        est.fit(X, y)
+    assert str(exc_info.value) == exc_msg, "Exception message does not match"
+
+
+@pytest.mark.parametrize('name', FOREST_CLASSIFIERS)
+def test_classification_toy_max_samples(name):
+    # Test that the toy example is separable via a bootstrap size of only 2
+
+    rng = np.random.RandomState(1)
+    max_tries = 100
+
+    # The toy example is separable using just one
+    # decision stump, and choosing 2 examples from the full
+    # 6-example dataset *if* the 2 examples are chosen correctly.
+    est = FOREST_CLASSIFIERS[name](
+        n_estimators=1,
+        bootstrap=True,
+        max_samples=2,
+        max_depth=1,
+        random_state=rng,
+    )
+
+    # Each call to fit uses a different bootstrap sample of size two. If we
+    # fit multiple times, we expect that we eventually hit a case where
+    # the two examples chosen for the bootstrap sample are from the opposite
+    # class and yield a perfect score across the entire dataset.
+    perfect_score = False
+    for _ in range(max_tries):
+        est.fit(X, y)
+        if est.score(X, y) == 1.0:
+            perfect_score = True
+            break
+
+    msg = "Perfect accuracy is achievable with `max_samples=2` on toy data"
+    assert perfect_score, msg
+
+
+@pytest.mark.parametrize('name', FOREST_CLASSIFIERS_REGRESSORS)
+def test_little_tree_with_small_max_samples(name):
```

Comment:
So apparently this test is only valid for the RandomForest (based on the DecisionTree). This not true for the ExtraTree

Reply:
Correct: https://github.com/scikit-learn/scikit-learn/pull/14682#discussion_r320058461

Reply:
Remove the tests for extra tree or remove the `max_samples` param for extra tree?

Reply:
Done in https://github.com/scikit-learn/scikit-learn/pull/14682/commits/9d72d36c2fe9ebb96892308b17cd1ae72809c555